shop-ui/src/main/java/com/maxshop/viewModel/ProductDetailsViewModel.kt

```diff
@@ -61,6 +66,27 @@ internal class ProductDetailsViewModel @Inject constructor(
         }
     }
 
+    fun addToBag() {
+        viewModelScope.launch {
+            _detailedProduct.value?.let {
+                try {
+                    addProductToBagUseCase.execute(
+                        detailedProductMapper.toBagProduct(
+                            it,
+                            if (it.colorsList == null) null
+                            else color.value
+                                ?: throw NullPointerException("Color can't be null"),
```

Comment:
In my view would be better to use `IllegalArgumentException`, coz there is no null pointer exception actually. There is a wrong argument